packages/components/src/internal/components/editable/EditableGridPanelForUpdate.tsx

```diff
@@ -33,54 +35,35 @@ interface Props {
     selectionData: Map<string, any>;
     setIsDirty?: (isDirty: boolean) => void;
     singularNoun?: string;
-    updateRows: (schemaQuery: SchemaQuery, rows: any[]) => Promise<any>;
+    updateRows: (schemaQuery: SchemaQuery, rows: any[], comment: string) => Promise<any>;
 }
 
-interface State {
-    dataModels: QueryModel[];
-    editorModels: EditorModel[];
-    error: string;
-    isSubmitting: boolean;
-    loaders: EditableGridLoader[];
-}
-
-export class EditableGridPanelForUpdate extends React.Component<Props, State> {
-    static defaultProps = {
-        singularNoun: 'row',
-        pluralNoun: 'rows',
-    };
-
-    constructor(props: Props) {
-        super(props);
-        const id = props.loader.id;
-
-        this.state = {
-            isSubmitting: false,
-            dataModels: [new QueryModel({ id, schemaQuery: props.queryModel.schemaQuery })],
-            editorModels: [new EditorModel({ id })],
-            error: undefined,
-            loaders: [props.loader],
-        };
-    }
-
-    componentDidMount(): void {
-        this.initEditorModel();
-    }
-
-    initEditorModel = async (): Promise<void> => {
-        const { dataModels, editorModels } = await initEditableGridModels(
-            this.state.dataModels,
-            this.state.editorModels,
-            this.state.loaders,
-            this.props.queryModel
-        );
-        this.setState({ dataModels, editorModels });
-    };
-
-    onGridChange: EditableGridChange = (event, editorModelChanges, dataKeys, data, index = 0): void => {
-        this.setState(state => {
-            const { dataModels, editorModels } = state;
-            return applyEditableGridChangesToModels(
+export const EditableGridPanelForUpdate: FC<Props> = props => {
```

Comment:
After reviewing it doesn't seem like this actually supports multiple editor models. I think this can be refactored to just be single model based.

```tsx
type Models = {
    dataModel: QueryModel;
    editorModel: EditorModel;
};

interface Props {
    containerFilter?: Query.ContainerFilter;
    getIsDirty?: () => boolean;
    idField: string;
    loader: EditableGridLoader;
    onCancel: () => void;
    onComplete: () => void;
    pluralNoun?: string;
    queryModel: QueryModel;
    selectionData: Map<string, any>;
    setIsDirty?: (isDirty: boolean) => void;
    singularNoun?: string;
    updateRows: (schemaQuery: SchemaQuery, rows: any[], comment: string) => Promise<any>;
}

export const EditableGridPanelForUpdate: FC<Props> = props => {
    const { containerFilter, onCancel, singularNoun, pluralNoun, ...editableGridProps } = props;
    const { idField, loader, queryModel, selectionData, getIsDirty, setIsDirty, updateRows, onComplete } =
        editableGridProps;
    const id = loader.id;

    const [models, setModels] = useState<Models>(() => ({
        dataModel: new QueryModel({ id, schemaQuery: queryModel.schemaQuery }),
        editorModel: new EditorModel({ id }),
    }));
    const [error, setError] = useState<string>();
    const [isSubmitting, setIsSubmitting] = useState<boolean>(false);
    const [comment, setComment] = useState<string>();
    const { requiresUserComment } = useDataChangeCommentsRequired();

    const hasValidUserComment = comment?.trim()?.length > 0;

    useEffect(() => {
        (async (): Promise<void> => {
            try {
                const models_ = await initEditableGridModel(models.dataModel, models.editorModel, loader, queryModel);
                setModels(models_);
            } catch (e) {
                console.error(e);
                setError(resolveErrorMessage(e) ?? 'Failed to initialize editable grid.');
            }
        })();
    }, []);

    const onGridChange: EditableGridChange = useCallback(
        (event, editorModelChanges, dataKeys, data, index = 0): void => {
            const { dataModels, editorModels } = applyEditableGridChangesToModels(
                [models.dataModel],
                [models.editorModel],
                editorModelChanges,
                undefined,
                dataKeys,
                data,
                index
            );

            const [dataModel] = dataModels;
            const [editorModel] = editorModels;
            setModels({ dataModel, editorModel });

            if (editorModel.isDataChangeEvent(event)) {
                setIsDirty?.(true);
            }
        },
        [models, setIsDirty]
    );

    const onSubmit = useCallback(async (): Promise<void> => {
        const gridData = getUpdatedDataFromEditableGrid(
            [models.dataModel],
            [models.editorModel],
            idField,
            selectionData
        );

        if (!gridData) {
            onComplete();
            return;
        }

        setIsSubmitting(true);

        try {
            await updateRows(gridData.schemaQuery, gridData.updatedRows, comment);
            setIsSubmitting(false);
            onComplete();
        } catch (e) {
            setError(e?.exception ?? 'There was a problem updating the ' + singularNoun + ' data.');
            setIsSubmitting(false);
        }
    }, [comment, models, idField, onComplete, selectionData, singularNoun, updateRows]);

    const onCommentChange = useCallback((_comment: string): void => {
        setComment(_comment);
    }, []);

    const columnMetadata = useMemo(
        () => getUniqueIdColumnMetadata(models.dataModel.queryInfo),
        [models.dataModel.queryInfo]
    );

    if (models.dataModel.isLoading) {
        return <LoadingSpinner />;
    }

    return (
        <>
            <EditableGridPanel
                {...editableGridProps}
                allowAdd={false}
                allowRemove={false}
                bordered
                bsStyle="info"
                columnMetadata={columnMetadata}
                containerFilter={containerFilter}
                editorModel={models.editorModel}
                forUpdate
                model={models.dataModel}
                onChange={onGridChange}
                striped
                title={`Edit selected ${pluralNoun}`}
            />
            <Alert>{error}</Alert>
            <WizardNavButtons
                cancel={onCancel}
                canFinish={getIsDirty?.() && (!requiresUserComment || hasValidUserComment)}
                nextStep={onSubmit}
                finish
                isFinishing={isSubmitting}
                isFinishingText="Updating..."
                isFinishedText="Finished Updating"
                finishText={
                    'Finish Updating ' +
                    models.dataModel.orderedRows.length +
                    ' ' +
                    (models.dataModel.orderedRows.length === 1
                        ? capitalizeFirstChar(singularNoun)
                        : capitalizeFirstChar(pluralNoun))
                }
            >
                <CommentTextArea
                    actionName="Update"
                    containerClassName="inline-comment"
                    onChange={onCommentChange}
                    requiresUserComment={requiresUserComment}
                    inline
                />
            </WizardNavButtons>
        </>
    );
};

EditableGridPanelForUpdate.defaultProps = {
    singularNoun: 'row',
    pluralNoun: 'rows',
};
```